From 8fd3d4d95e565a27657092f5bb678815375fcfb5 Mon Sep 17 00:00:00 2001
From: darealshinji <djcj@gmx.de>
Date: Thu, 5 Nov 2015 18:57:21 +0100
Subject: [PATCH 20/79] fixing an argument parsing bug in ytasm

Taken from a Debian patch.
More information can be found at
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=716639
http://anonscm.debian.org/viewvc/sam-hocevar/pkg-misc/unstable/yasm/debian/patches/100_ytasm_parsing_error.diff?view=log
---
 frontends/tasm/tasm-options.c | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/frontends/tasm/tasm-options.c b/frontends/tasm/tasm-options.c
index 57c3bdfd..3d40f31b 100644
--- a/frontends/tasm/tasm-options.c
+++ b/frontends/tasm/tasm-options.c
@@ -65,15 +65,18 @@ parse_cmdline(int argc, char **argv, opt_option *options, size_t nopts,
 
                     param = &argv[0][1+len];
                     if (options[i].takes_param) {
-                        if (param[0] == '\0') {
+                        if (param[0] != '\0') {
+                            /* do nothing */
+                        } else if (argc > 1) {
+                            param = argv[1];
+                            argc--;
+                            argv++;
+                        } else {
                             print_error(
-                                _("option `-%c' needs an argument!"),
+                                _("option `-%s' needs an argument!"),
                                 options[i].opt);
                             errors++;
                             goto fail;
-                        } else {
-                            argc--;
-                            argv++;
                         }
                     } else
                         param = NULL;
-- 
2.49.0

